roles/docker_private_registry/tasks/main.yaml

```diff
@@ -0,0 +1,38 @@
+---
+#
+#  This role sets up and creates an insecure private registry on the localhost
+#  at port 5000. 
+#
+  - name: Fail if ansible_docker0.ipv4.address is not defined
+    fail: 
+      msg: "ansible_docker0.ipv4.address it not defined!"
+    when: ansible_docker0.ipv4.address is not defined
+    run_once: true
+
+  - name: edit /etc/sysconfig/docker file to add private registry
+    replace: 
+      dest=/etc/sysconfig/docker 
+      regexp='^ADD_REGISTRY=\'--add-registry registry.access.redhat.com\'' 
+      replace='ADD_REGISTRY=\'--add-registry {{ ansible_docker0.ipv4.address }}:5000 --add-registry registry.access.redhat.com\''
+
+  - name: edit /etc/sysconfig/docker file to enable insecure registry
+    replace:
+      dest=/etc/sysconfig/docker
+      regexp='^# INSECURE_REGISTRY=\'--insecure-registry\s?\''
+      replace='INSECURE_REGISTRY=\'--insecure-registry {{ ansible_docker0.ipv4.address }}:5000\''
+
+  - name: restart docker
+    service: name=docker state=restarted
+
+  - name: wait for docker restart to complete
+    pause: seconds=30
```

Comment:
I changed this to stop and start docker using the service module and the sleep is no longer needed.  With only restart, the next step (creating a private docker registry) would fail.

I also tried to remove the pause after by stopping and starting the kubernetes and etcd services but I am getting connection refused from etcd creating the replication controller and services.

failed: [rhelah] (item=db-rc.yaml) => {"changed": true, "cmd": ["kubectl", "create", "-f", "/root/db-rc.yaml"], "delta": "0:00:00.038802", "end": "2016-06-06 15:05:40.701356", "failed": true, "item": "db-rc.yaml", "rc": 1, "start": "2016-06-06 15:05:40.662554", "stderr": "The connection to the server localhost:8080 was refused - did you specify the right host or port?", "stdout": "", "stdout_lines": [], "warnings": []}
failed: [rhelah] (item=webserver-rc.yaml) => {"changed": true, "cmd": ["kubectl", "create", "-f", "/root/webserver-rc.yaml"], "delta": "0:00:00.036192", "end": "2016-06-06 15:05:41.070615", "failed": true, "item": "webserver-rc.yaml", "rc": 1, "start": "2016-06-06 15:05:41.034423", "stderr": "The connection to the server localhost:8080 was refused - did you specify the right host or port?", "stdout": "", "stdout_lines": [], "warnings": []}